### ISSUE TITLE
`jsdoc/require-returns-check`: Support the `never` type

### ISSUE THREAD

**Issue by stof** [gajus/eslint-plugin-jsdoc#817]
## Expected behavior
Using the `never` return type on a method not using `return` statement should be allowed (it could even check that you don't return).

The `never` type is not equivalent to the `void` type (which is what the absence of a `@returns` represents). So being able to use the `never` type is useful when using type-checking (through `tsc --allow-js --check-js`) on the JS code

## Actual behavior
The plugin rule complains about the presence of the `@returns` tag

## ESLint Config

```json
{
  "root": true,
  "plugins": ["jsdoc"],
  "rules": {
    "jsdoc/require-returns-check": "error"
  } 
}
```

## ESLint sample

```js
class MyMetric {
  /**
   * @returns {never}
   */
  computeByUsers() {
    throw new Error('Computation grouped by user is not supported for this metric');
  }
}
```

## Environment

- Node version: 14.18.2
- ESLint version: 7.32.0
- `eslint-plugin-jsdoc` version: 37.1.0

**Comment by brettz9** [comment]
Although this has been effectively closed in https://github.com/gajus/eslint-plugin-jsdoc/releases/tag/v37.2.8 , you might be interested to note that I've filed #819 to confirm that a present `never` is not only allowed but checks that a corresponding return is present.

**Comment by stof** [comment]
thanks for the quick fix
